Source/WebKit2/Platform/IPC/Connection.cpp

```diff
@@ -873,6 +873,7 @@ void Connection::didFailToSendSyncMessage()
     if (!m_shouldExitOnSyncMessageSendFailure)
         return;
 
+    m_client.didClose(*this);
     exit(0);
```

Comment:
didClose normally stops the run loop, so maybe we don't need to exit this way anymore here.